api/client/inspect.go

```diff
@@ -101,42 +100,45 @@ func (cli *DockerCli) CmdInspect(args ...string) error {
 		} else {
 			rdr := bytes.NewReader(obj)
 			dec := json.NewDecoder(rdr)
+			buf := bytes.NewBufferString("")
 
 			if isImage {
 				inspPtr := types.ImageInspect{}
 				if err := dec.Decode(&inspPtr); err != nil {
-					fmt.Fprintf(cli.err, "%s\n", err)
+					fmt.Fprintf(cli.err, "Unable to read inspect data: %v\n", err)
 					status = 1
-					continue
+					break
```

Comment:
The previous behavior explained like a movie:

- Client: hey, can I have this container's info?
- API: sure, here you have some gibberish that might not even be a json payload
- Client: cool, can I have this other container's info?

The new behavior goes like this:

- Client: hey, can I have this container's info?
- API: sure, here you have some gibberish that might not even be a json payload
- Client: what's wrong with you API?? abort!